internal/compiler/parser/type.rs

```diff
@@ -85,3 +85,89 @@ pub fn parse_struct_declaration(p: &mut impl Parser) -> bool {
     parse_type_object(&mut *p);
     true
 }
+
+pub fn parse_rustattr(p: &mut impl Parser) -> bool {
+    debug_assert_eq!(p.peek().as_str(), "@");
+    let mut p = p.start_node(SyntaxKind::RustAttr);
+    p.consume(); // "@"
+    if p.peek().as_str() != "rust-attr" {
+        p.expect(SyntaxKind::RustAttr);
+    }
+    p.consume(); // "rust-attr"
+    p.expect(SyntaxKind::LParent);
+    parse_deriven(&mut *p);
+    if p.peek().as_str() == "export" {
+        p.consume();
+    }
+    parse_struct_declaration(&mut *p);
+    true
+}
+
+macro_rules! deriven_feature {
+    () => {
+        ["serde"]
+    };
+}
+
+macro_rules! deriven_macro_trait {
+    ($feature:expr) => {
+        match $feature {
+            "serde" => vec!["Serialize", "Deserialize"],
+            _ => vec![],
+        }
+    };
+}
+
+pub fn parse_deriven(p: &mut impl Parser) -> bool {
```

Comment:
I would actually just treat the inside of a @rust-attr  as some opaque token, and just have these token be replicated verbatim in the generated rust.

This has the advantage of being generic, but has the problem that compilation errors might be harder to track to their source.  But I think for this kind of feature, we really want to support anything.
Such as `@rust-attr(derive(serde::Serialize, serde::Deserialize))`  or even `@rust-attr(doc = "whatever")`

Reply:
How can we validate it ?

Reply:
> How can we validate it ? Should we validate it ? 

I think it can just be an arbitrary balanced token tree, so we can just validate that it has the right amount of parentheses and braces.

> And other thing is it would be used as String , can we use String as evaluate Statement in quote! ?

`TokenStream` implements `FromStr`